### ISSUE TITLE
Pod calico-node stuck in CrashLoopBackOff status

### ISSUE THREAD

**Issue by josephrodriguez** [projectcalico/calico#8631]
Unable to complete installation of latest version of Calico for Kubernetes 1.29.3 from:

https://docs.tigera.io/calico/latest/getting-started/kubernetes/self-managed-onprem/onpremises#install-calico.

The pod has this error:
```

calico-node: error while loading shared libraries: libpcap.so.0.8: cannot open shared object file: No such file or directory
Calico node failed to start
```

## Expected Behavior

All the pods for the Deployments: `calico-kube-controllers` and `calico-typha ` , DaemonSets `calico-node` and `csi-node-driver` start to run normally.

## Current Behavior

The calico-node Pod is stuck  in `CrashLoopBackOff` status unable to start.
```
$ kubectl get pod -A

NAMESPACE         NAME                                       READY   STATUS              RESTARTS        AGE
calico-system     calico-kube-controllers-5d9485c555-rz2mz   0/1     ContainerCreating   0               26m
calico-system     calico-node-fnr2x                          0/1     CrashLoopBackOff    6 (3m45s ago)   9m36s
calico-system     calico-typha-5b9df48655-xsr4p              1/1     Running             1 (16m ago)     26m
calico-system     csi-node-driver-b6zzz                      0/2     ContainerCreating   0               26m
kube-system       coredns-76f75df574-726np                   0/1     ContainerCreating   0               34m
kube-system       coredns-76f75df574-bk5k5                   0/1     ContainerCreating   0               34m
kube-system       etcd-master                                1/1     Running             7 (16m ago)     34m
kube-system       kube-apiserver-master                      1/1     Running             7 (16m ago)     34m
kube-system       kube-controller-manager-master             1/1     Running             3 (16m ago)     34m
kube-system       kube-proxy-w64xl                           1/1     Running             1 (16m ago)     34m
kube-system       kube-scheduler-master                      1/1     Running             7 (16m ago)     34m
tigera-operator   tigera-operator-748c69cf45-wf7ht           1/1     Running             1 (16m ago)     26m
```

## Steps to Reproduce (for bugs)

Following the steps:

1. Install CNI:
```
$ curl -LOs https://github.com/containernetworking/plugins/releases/download/v1.4.1/cni-plugins-linux-arm64-v1.4.1.tgz
$ sudo mkdir -p /opt/cni/bin
$ sudo tar Cxzvf /opt/cni/bin cni-plugins-linux-arm64-v1.4.1.tgz
```

2. Initialize kubeadm:

```
kubeadm init  --pod-network-cidr=10.32.0.0/16
```

3. Install Calico according the [documentation](https://docs.tigera.io/calico/latest/getting-started/kubernetes/self-managed-onprem/onpremises#install-calico):

```
$ kubectl create -f https://raw.githubusercontent.com/projectcalico/calico/v3.27.2/manifests/tigera-operator.yaml
$ curl https://raw.githubusercontent.com/projectcalico/calico/v3.27.2/manifests/custom-resources.yaml -O

Update CALICO_IPV4POOL_CIDR variable to used pod CIDR: 10.32.0.0/16 and uncomment the env variable.

$ kubectl create -f custom-resources.yaml
```

This is the definition of calico-node Pod:

```
apiVersion: v1
kind: Pod
metadata:
  annotations:
    hash.operator.tigera.io/cni-config: 7c3d4f43aed6c9f83376215a5e9a368bef0902a0
    hash.operator.tigera.io/system: fdde45054a8ae4f629960ce37570929502e59449
    tigera-operator.hash.operator.tigera.io/tigera-ca-private: df2ea845b241ce8f4f6ef9d3767528f5806f943e
  creationTimestamp: "2024-03-19T17:15:49Z"
  generateName: calico-node-
  labels:
    app.kubernetes.io/name: calico-node
    controller-revision-hash: 74d8cc9599
    k8s-app: calico-node
    pod-template-generation: "1"
  name: calico-node-fnr2x
  namespace: calico-system
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: DaemonSet
    name: calico-node
    uid: a0b48ec3-0d2b-4c91-a18a-86d64ebfd466
  resourceVersion: "6548"
  uid: dd4ae0bd-ba58-4034-b816-73f422cf5953
spec:
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchFields:
          - key: metadata.name
            operator: In
            values:
            - master
  containers:
  - env:
    - name: DATASTORE_TYPE
      value: kubernetes
    - name: WAIT_FOR_DATASTORE
      value: "true"
    - name: CLUSTER_TYPE
      value: k8s,operator,bgp
    - name: CALICO_DISABLE_FILE_LOGGING
      value: "false"
    - name: FELIX_DEFAULTENDPOINTTOHOSTACTION
      value: ACCEPT
    - name: FELIX_HEALTHENABLED
      value: "true"
    - name: FELIX_HEALTHPORT
      value: "9099"
    - name: NODENAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: spec.nodeName
    - name: NAMESPACE
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.namespace
    - name: FELIX_TYPHAK8SNAMESPACE
      value: calico-system
    - name: FELIX_TYPHAK8SSERVICENAME
      value: calico-typha
    - name: FELIX_TYPHACAFILE
      value: /etc/pki/tls/certs/tigera-ca-bundle.crt
    - name: FELIX_TYPHACERTFILE
      value: /node-certs/tls.crt
    - name: FELIX_TYPHAKEYFILE
      value: /node-certs/tls.key
    - name: FIPS_MODE_ENABLED
      value: "false"
    - name: FELIX_TYPHACN
      value: typha-server
    - name: CALICO_MANAGE_CNI
      value: "true"
    - name: CALICO_IPV4POOL_CIDR
      value: 10.32.0.0/16
    - name: CALICO_IPV4POOL_VXLAN
      value: CrossSubnet
    - name: CALICO_IPV4POOL_BLOCK_SIZE
      value: "26"
    - name: CALICO_IPV4POOL_NODE_SELECTOR
      value: all()
    - name: CALICO_IPV4POOL_DISABLE_BGP_EXPORT
      value: "false"
    - name: CALICO_NETWORKING_BACKEND
      value: bird
    - name: IP
      value: autodetect
    - name: IP_AUTODETECTION_METHOD
      value: first-found
    - name: IP6
      value: none
    - name: FELIX_IPV6SUPPORT
      value: "false"
    - name: KUBERNETES_SERVICE_HOST
      value: 10.96.0.1
    - name: KUBERNETES_SERVICE_PORT
      value: "443"
    image: docker.io/calico/node:v3.27.2
    imagePullPolicy: IfNotPresent
    lifecycle:
      preStop:
        exec:
          command:
          - /bin/calico-node
          - -shutdown
    livenessProbe:
      failureThreshold: 3
      httpGet:
        host: localhost
        path: /liveness
        port: 9099
        scheme: HTTP
      periodSeconds: 60
      successThreshold: 1
      timeoutSeconds: 10
    name: calico-node
    readinessProbe:
      exec:
        command:
        - /bin/calico-node
        - -bird-ready
        - -felix-ready
      failureThreshold: 3
      periodSeconds: 30
      successThreshold: 1
      timeoutSeconds: 10
    resources: {}
    securityContext:
      allowPrivilegeEscalation: true
      capabilities:
        drop:
        - ALL
      privileged: true
      runAsGroup: 0
      runAsNonRoot: false
      runAsUser: 0
      seccompProfile:
        type: RuntimeDefault
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /etc/pki/tls/certs
      name: tigera-ca-bundle
      readOnly: true
    - mountPath: /etc/pki/tls/cert.pem
      name: tigera-ca-bundle
      readOnly: true
      subPath: ca-bundle.crt
    - mountPath: /lib/modules
      name: lib-modules
      readOnly: true
    - mountPath: /run/xtables.lock
      name: xtables-lock
    - mountPath: /var/run/nodeagent
      name: policysync
    - mountPath: /node-certs
      name: node-certs
      readOnly: true
    - mountPath: /var/run/calico
      name: var-run-calico
    - mountPath: /var/lib/calico
      name: var-lib-calico
    - mountPath: /var/log/calico/cni
      name: cni-log-dir
    - mountPath: /host/etc/cni/net.d
      name: cni-net-dir
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: kube-api-access-ldj6w
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  hostNetwork: true
  initContainers:
  - image: docker.io/calico/pod2daemon-flexvol:v3.27.2
    imagePullPolicy: IfNotPresent
    name: flexvol-driver
    resources: {}
    securityContext:
      allowPrivilegeEscalation: true
      capabilities:
        drop:
        - ALL
      privileged: true
      runAsGroup: 0
      runAsNonRoot: false
      runAsUser: 0
      seccompProfile:
        type: RuntimeDefault
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /host/driver
      name: flexvol-driver-host
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: kube-api-access-ldj6w
      readOnly: true
  - command:
    - /opt/cni/bin/install
    env:
    - name: CNI_CONF_NAME
      value: 10-calico.conflist
    - name: SLEEP
      value: "false"
    - name: CNI_NET_DIR
      value: /etc/cni/net.d
    - name: CNI_NETWORK_CONFIG
      valueFrom:
        configMapKeyRef:
          key: config
          name: cni-config
    - name: KUBERNETES_SERVICE_HOST
      value: 10.96.0.1
    - name: KUBERNETES_SERVICE_PORT
      value: "443"
    image: docker.io/calico/cni:v3.27.2
    imagePullPolicy: IfNotPresent
    name: install-cni
    resources: {}
    securityContext:
      allowPrivilegeEscalation: true
      capabilities:
        drop:
        - ALL
      privileged: true
      runAsGroup: 0
      runAsNonRoot: false
      runAsUser: 0
      seccompProfile:
        type: RuntimeDefault
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /host/opt/cni/bin
      name: cni-bin-dir
    - mountPath: /host/etc/cni/net.d
      name: cni-net-dir
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: kube-api-access-ldj6w
      readOnly: true
  nodeName: master
  nodeSelector:
    kubernetes.io/os: linux
  preemptionPolicy: PreemptLowerPriority
  priority: 2000001000
  priorityClassName: system-node-critical
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: calico-node
  serviceAccountName: calico-node
  terminationGracePeriodSeconds: 5
  tolerations:
  - key: CriticalAddonsOnly
    operator: Exists
  - effect: NoSchedule
    operator: Exists
  - effect: NoExecute
    operator: Exists
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/disk-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/memory-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/pid-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/unschedulable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/network-unavailable
    operator: Exists
  volumes:
  - hostPath:
      path: /lib/modules
      type: ""
    name: lib-modules
  - hostPath:
      path: /run/xtables.lock
      type: FileOrCreate
    name: xtables-lock
  - hostPath:
      path: /var/run/nodeagent
      type: DirectoryOrCreate
    name: policysync
  - configMap:
      defaultMode: 420
      name: tigera-ca-bundle
    name: tigera-ca-bundle
  - name: node-certs
    secret:
      defaultMode: 420
      secretName: node-certs
  - hostPath:
      path: /var/run/calico
      type: ""
    name: var-run-calico
  - hostPath:
      path: /var/lib/calico
      type: ""
    name: var-lib-calico
  - hostPath:
      path: /opt/cni/bin
      type: ""
    name: cni-bin-dir
  - hostPath:
      path: /etc/cni/net.d
      type: ""
    name: cni-net-dir
  - hostPath:
      path: /var/log/calico/cni
      type: ""
    name: cni-log-dir
  - hostPath:
      path: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds
      type: DirectoryOrCreate
    name: flexvol-driver-host
  - name: kube-api-access-ldj6w
    projected:
      defaultMode: 420
      sources:
      - serviceAccountToken:
          expirationSeconds: 3607
          path: token
      - configMap:
          items:
          - key: ca.crt
            path: ca.crt
          name: kube-root-ca.crt
      - downwardAPI:
          items:
          - fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
            path: namespace
```

- /etc/cni/net.d/10-calico.conflist

```
{
  "name": "k8s-pod-network",
  "cniVersion": "0.3.1",
  "plugins": [
    {
      "container_settings": {
        "allow_ip_forwarding": false
      },
      "datastore_type": "kubernetes",
      "ipam": {
        "assign_ipv4": "true",
        "assign_ipv6": "false",
        "type": "calico-ipam"
      },
      "kubernetes": {
        "k8s_api_root": "https://10.96.0.1:443",
        "kubeconfig": "/etc/cni/net.d/calico-kubeconfig"
      },
      "log_file_max_age": 30,
      "log_file_max_count": 10,
      "log_file_max_size": 100,
      "log_file_path": "/var/log/calico/cni/cni.log",
      "log_level": "Info",
      "mtu": 0,
      "nodename_file_optional": false,
      "policy": {
        "type": "k8s"
      },
      "type": "calico"
    },
    {
      "capabilities": {
        "bandwidth": true
      },
      "type": "bandwidth"
    },
    {
      "capabilities": {
        "portMappings": true
      },
      "snat": true,
      "type": "portmap"
    }
  ]
}
```

## Your Environment

* Calico version: 3.27.2
* Orchestrator version: Kubernetes (on-premise) 1.29.3
* Operating System and version: `Linux master 6.6.20+rpt-rpi-v8 #1 SMP PREEMPT Debian 1:6.6.20-1+rpt1 (2024-03-07) aarch64 GNU/Linux`

**Comment by bootc** [comment]
This is a duplicate of #8541.

**Comment by josephrodriguez** [comment]
Fixed downgrading to version v3.27.0.
